### ISSUE TITLE
Create LongHistogram 

### ISSUE THREAD

**Issue by LeeCampbell** [HdrHistogram/HdrHistogram.NET#4]
Create the default implementation with signed 64 bit counts.

It should support
 * auto-sizing
 * percentile, linear, and log based iteration
 * encode into the compressed V2 serialized histogram format (which is what the current Java code will encode to, as well as the current C and Python code bases).
 * decode V2. 
 * SingleWriter i.e. not thread safe for writes.

I also propose that the current `Histogrram` type be renamed to `LongHistogram`. The `Histogram` type can then be used as a factory that can help guide users to which instance they need.

potentially something like 

```
var histogram = Histogram
    .With64bitBucketCount() 
    .AsThreadSafe()
    .Create()
```

**Comment by mattwarren** [comment]
+1 for the idea of having a factory with a fluent api, it's a nice idea

**Comment by LeeCampbell** [comment]
closed by #14

Factory feature moved to #16
